Source/engine/random.cpp

```diff
@@ -51,4 +51,9 @@ int32_t GenerateRnd(int32_t v)
 	return AdvanceRndSeed() % v;
 }
 
+bool FlipCoin()
+{
+	return GenerateRnd(2) != 0;
+}
```

Comment:
Structuring this check to be `== 0` is more natural, especially when dealing with frequencies other than 1 in 2. To fully address #2193 you could make this function take a frequency parameter with a default of 2. Then you can replace calls to GenerateRnd() like items.cpp:1084,1156,1157,1167,1224,1600, objects.cpp:504,538,542,546,550, etc.

Reply:
See https://github.com/diasurgical/devilutionX/pull/4908#discussion_r916160846 for why we flipped it. Not that we couldn't just add a lot of `!` instead.

Reply:
Opened #4913 with the result of inverting the condition, can adjust most of the callers to suit :D